src/storage/dbworker.cpp

```diff
@@ -550,13 +550,13 @@ void DBWorker::getHistory(const QString &filter)
 
     if (!filter.isEmpty()) {
         filterQuery = filterQuery.arg(QString("(url LIKE :search OR title LIKE :search)"));
-        order = QString("visited_count DESC, date ASC, LENGTH(url), title");
+        order = QString("visited_count ASC, date DESC, LENGTH(url), title");
```

Comment:
Both previous and this are off. For both visited_count and date we should have DESC ordering so that most visited would be still preferred. That said, problem is that visited count is take higher presendence than date (most recently visited).

Hence, I'd say that this actually should be like: `order = QString("date DESC, visited_count DESC, LENGTH(url), title");`

Reply:
While you're at this could you fix also 453 that is behind debug logging: "qDebug() << "link id:" << linkId;"